fab/utils.py

```diff
@@ -102,11 +121,38 @@ def _get_github():
         ).format(project_root=PROJECT_ROOT)
         username = raw_input('Github username: ')
         password = getpass('Github password: ')
-        return login(
+        global_github = login(
             username=username,
             password=password,
         )
     else:
-        return login(
+        global_github = login(
             token=GITHUB_APIKEY,
         )
+
+    return global_github
+
+
+def cache_deploy_state(command_index):
+    with open(os.path.join(PROJECT_ROOT, CACHED_DEPLOY_CHECKPOINT_FILENAME), 'w') as f:
+        pickle.dump(command_index, f)
+    with open(os.path.join(PROJECT_ROOT, CACHED_DEPLOY_ENV_FILENAME), 'w') as f:
+        pickle.dump(env, f)
+
+
+def retrieve_cached_deploy_env():
+    filename = os.path.join(PROJECT_ROOT, CACHED_DEPLOY_ENV_FILENAME)
+    return _retrieve_cached(filename)
+
+
+def retrieve_cached_deploy_checkpoint():
+    filename = os.path.join(PROJECT_ROOT, CACHED_DEPLOY_CHECKPOINT_FILENAME)
+    return _retrieve_cached(filename)
+
+
+def _retrieve_cached(filename):
+    with open(filename, 'r') as f:
+        try:
+            return pickle.load(f)
+        except Exception:
```

Comment:
Do you actually want to catch this exception?  I can't imagine we'd expect the pickled file to be corrupt or otherwise unreadable, might be good to fail hard if this occurs.